lib/dal/src/audit_logging.rs

```diff
@@ -166,3 +164,151 @@ pub(crate) async fn write_final_message(ctx: &DalContext) -> AuditLoggingResult<
         .await?;
     Ok(())
 }
+
+#[instrument(name = "audit_logging.list", level = "debug", skip_all)]
+pub async fn list(ctx: &DalContext) -> Result<Vec<FrontendAuditLog>> {
+    // TODO(nick): nuke this from intergalactic orbit. Then do it again.
+    let workspace_id = match ctx.workspace_pk() {
+        Ok(workspace_id) => workspace_id,
+        Err(TransactionsError::Tenancy(TenancyError::NoWorkspace)) => return Ok(Vec::new()),
+        Err(err) => return Err(AuditLoggingError::Transactions(Box::new(err))),
+    };
+
+    let stream_wrapper = AuditLogsStream::get_or_create(ctx.jetstream_context()).await?;
+    let stream = stream_wrapper.stream().await?;
+    let consumer = stream
+        .create_consumer(async_nats::jetstream::consumer::pull::Config {
+            filter_subject: stream_wrapper.subject(workspace_id.into()).to_string(),
+            ..Default::default()
+        })
+        .await?;
+
+    // TODO(nick): remove hard-coded max messages value.
+    let mut messages = consumer.fetch().max_messages(200).messages().await?;
```

Comment:
Yeah, maybe one day an alternative `list_by_chunk` function that takes the size of messages and maybe a sequence number (or similar) as the cursor. I'm guessing you're working into a paging strategy...

Reply:
Exactly. We have a "fake" paging strategy now, which is to dump a ton of logs and then simulate paging. I am also considering switching from this standard, non-durable pull consumer to an ordered push consumer that can perform this behavior. Right now, we're not so focused on the NATS consumption pieces and are focused on the frontend experience, but this will become important very soon.